### DISCUSSION TITLE
What are your most urgent features in GDL ? (missing or weak or slow)

### DISCUSSION THREAD

**Discussion by alaingdl** [gnudatalanguage/gdl#1677] [General]
I think it is time to discuss what is weak or missing in GDL now. On my side most of the functionalities are working fine & fast, but yes, basically I use basic operations on data cube + The Astro Lib (FITS I/O) + MPfit

Missing : 
- #860 SHMMAP and SHMVAR
- #1605 some keywords give bad outputs in SHADE_SURF
- #1389 missing FILE keyword in LOADCT 

Weak :
- better communication on the good quality of current GDL (the 1.0.4 !)
- since IDL 8.5 we are slower in some matrix operations (eg INVERT())
- #1676 UFT-8 related (should be possible because PLplot & WxWidgets are suppose to handle)
- some operations are slow on M1 & M2 (why ? only compiler issue ?!)
- when compiled with -Wall, huge amounts of warnings ... 
- graphical tests ? (no provocation, just to mention we have to time time and systematize that)
- automatic tests on the PS outputs (no provocation, just to mention we have to time time and systematize that)
-  CHOLDC/CHOLSOL exact behavior (partially OK) + tests are missing #1311 

Bugs :
- #1521 SURFACE & SHADE_SURF + !p.multi
- #1519 TIMESTAMP (I have a draft somewhere, but few tricky bugs remain :sad:)
- #1413 bug(s) or detail in POLY_FIT & test_poly_fit
- #230  TRIRED related family functions
- ... (I did not review all the issues ! Some may be closed)

To be discussed, corrected and continued ...

Alain

**Comment by slayoo** [comment]
I'd also prioritise bringing back execution of Python-GDL tests contributed by @olebole on CI (this is a regression since we moved from Travis to GHA), see: https://github.com/gnudatalanguage/gdl/issues/1055

**Comment by GillesDuvert** [comment]
At the moment I'm experimenting save/restore of procedures. No need to emphasise the interest.

Shmmap should be next as it's easy.

**Comment by GillesDuvert** [comment]
If gdl  becomes handy, then the gdlde and the jupyter interest should be revived. Hope we'll get new contributors.
